core/src/test/java/org/apache/brooklyn/util/core/task/ValueResolverTest.java

```diff
@@ -41,6 +49,125 @@ public void setUp() throws Exception {
         super.setUp();
     }
     
+    public void testTimeoutZero() {
+        Maybe<String> result = Tasks.resolving(newSleepTask(Duration.TEN_SECONDS, "foo")).as(String.class).context(app).timeout(Duration.ZERO).getMaybe();
+        Assert.assertFalse(result.isPresent());
+    }
+    
+    public void testTimeoutBig() {
+        Maybe<String> result = Tasks.resolving(newSleepTask(Duration.ZERO, "foo")).as(String.class).context(app).timeout(Duration.TEN_SECONDS).getMaybe();
+        Assert.assertEquals(result.get(), "foo");
+    }
+
+    public void testNoExecutionContextOnCompleted() {
```

Comment:
Don't follow what this is testing - how is it testing something about 'no execution context'?